CONTRIBUTING.md

```diff
@@ -30,3 +30,41 @@ Use culturally neutral language:
 
   What you may consider well-known may be entirely obscure and distracting to people from different backgrounds.
 
+### Terminology
+
+To avoid confusion around all the things called Nix, always use the following terms with capitalization as given.
+
+Terms in brackets (`[  ]`) are optional.
+Terms in parentheses (`(  )`) are abbreviations.
+
+- Nix ecosystem
+
+  All of the below.
+
+  - Nix package manager
+
+    - `nix` [command [line interface]] (Nix CLI)
+
+  - Nix language
+
+      Programming langauge to declare packages and configurations for the Nix package manager.
+
+    - Nix expression
+
+      Expression written in Nix language.
+
+    - Nix file
+
+      File containing Nix expression, usally ends in `.nix`.
+
+  - Nix package collection (`nixpkgs`)
+
+    Read `nixpkgs` as "Nix packages".
+
+  - NixOS [Linux distribution]
+
+    Linux distribution based on Nix package manager and `nixpkgs`.
+    Read /nɪks oʊ ɛs/ ("Nix Oh Es").
+
+    - NixOS module collection (NixOS modules)
```

Comment:
I would introduce:

- The `NixOS/nixpkgs`-Repository implements:
  - Nix Package Collection &mdash; `./pkgs`
  - NixOS Module Collection &mdash; under `./nixos`
  - Nix Library Collection &mdash; `./lib` (also inndividually as smaller `nix-community/nixpkgs.lib`)

Reply:
I would explicitly call-out the misnomer. Not addressing this upfront makes people question their own best judgement at some point which is unpleasant.

Reply:
> The `NixOS/nixpkgs`-Repository implements:

I follow your thought but I think your proposal gets it backwards. The goal right now is to give things a sufficiently distinguishable name which is not misleading. It should not matter where these things live (it may change), but what they are.